benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/tasks/running/CheckTrialResultTask.java

```diff
@@ -67,22 +71,23 @@ public TrialResult call() throws Exception {
 
       }
 
-      if (trialResult == TrialResult.FAILURE && retries >= maxRetries) {
-        trialResult = TrialResult.EXECUTION_FAILURE;
+      if (failureStatus == null) {
```

Comment:
I think this anyway should happen in the controller, because you have to drive the control flow according to the status you get